src/api/app/models/review.rb

```diff
@@ -17,6 +17,16 @@ class NotFoundError < APIException
   validates :reason, length: { maximum: 65534 }
 
   validate :check_initial, on: [:create]
+  validate :validate_recursive
+
+  belongs_to :assigned_from, class_name: 'Review', foreign_key: :review_id
+  has_one :assigned_to, class_name: 'Review', foreign_key: :review_id
```

Comment:
Also maybe can we make these associations be called something like `review_assigned_from` and `review_assigned_to`? I think it makes it clearer to somebody looking at the first time that when they see `review.review_assigned_from` that this is self-to-self association. Otherwise when I first saw `review.assigned_from` I thought it was probably referencing a user or something like that.. What do you think?